tests/test_result.py

```diff
@@ -265,3 +291,15 @@ def f(a: int) -> int:
     res: Result[int, ValueError]
     res = f(123)  # No mypy error here.
     assert res.ok() == 123
+
+
+def sq(i: int) -> Result[int, int]:
+    return Ok(i * i)
+
+
+def to_err(i: int) -> Result[int, int]:
+    return Err(i)
+
+
+sq_anon: Callable[[int], Result[int, int]] = lambda i: Ok(i * i)
```

Comment:
why the lambdas? it looks like a very convoluted way to rewrite the `sq` helper above. is this to stress test type checkers? if so, please add a comment why/how 🙏🏼

Reply:
Yes, I meant it as a means of making sure we have lambdas and regular functions covered and both don't have any type checking issues. Thanks for pointing this out, I'll add a comment.